src/Microsoft.DotNet.ProjectModel/ProjectSystem/ProjectSystemCache.cs

```diff
@@ -0,0 +1,203 @@
+                {
+                    mainProject = (ProjectDescription)library;
+                    continue;
+                }
+
+                // resolving project reference
+                var projectLibrary = (ProjectDescription)library;
+                var targetFrameworkInformation = projectLibrary.TargetFrameworkInfo;
+
+                // if this is an assembly reference then treat it like a file reference
+                if (!string.IsNullOrEmpty(targetFrameworkInformation?.AssemblyPath) &&
+                     string.IsNullOrEmpty(targetFrameworkInformation?.WrappedProject))
+                {
+                    var assemblyPath = Path.GetFullPath(
+                        Path.Combine(projectLibrary.Project.ProjectDirectory,
+                                     targetFrameworkInformation.AssemblyPath));
+                    runtimeAssemblyReferences.Add(assemblyPath);
+                }
+                else
+                {
+                    string wrappedProjectPath = null;
+                    if (!string.IsNullOrEmpty(targetFrameworkInformation?.WrappedProject) &&
+                         projectLibrary.Project == null)
+                    {
+                        wrappedProjectPath = Path.GetFullPath(
+                            Path.Combine(projectLibrary.Project.ProjectDirectory,
+                                         targetFrameworkInformation.AssemblyPath));
+                    }
+
+                    projectReferences.Add(new ProjectReferenceInfo
+                    {
+                        Name = library.Identity.Name,
+                        Framework = library.Framework,
+                        Path = library.Path,
+                        WrappedProjectPath = wrappedProjectPath
+                    });
+                }
+            }
+
+            var exporter = new LibraryExporter(mainProject, projectContext.LibraryManager, configuration);
+            var exports = exporter.GetAllExports();
+
+            foreach (var export in exporter.GetAllExports())
+            {
+                runtimeAssemblyReferences.AddRange(export.CompilationAssemblies.Select(asset => asset.ResolvedPath));
+                exportedSourceFiles.AddRange(export.SourceReferences);
+            }
+
+            return new DependencyInfo(diagnostics,
+                                      dependencies,
+                                      projectReferences,
+                                      runtimeAssemblyReferences,
+                                      exportedSourceFiles);
+        }
+
+        private DependencyDescription CreateDependencyDescription(
+            LibraryDescription library,
+            ILookup<string, LibraryDescription> librariesLookup,
+            IEnumerable<DiagnosticMessage> diagnostics)
+        {
+            return new DependencyDescription
+            {
+                Name = library.Identity.Name,
+                // TODO: Correct? Different from DTH
```

Comment:
Yes, this should be fine. DisplayName and Name should be the same in the "new" DTH. This was to cover the renaming of framework assemblies, where we renamed framework references to "fx/[name]" in order to avoid circular reference errors. We aren't doing that in the new code.